### ISSUE TITLE
[OneDNN] Error "could not create a primitive" on AMD CPU

### ISSUE THREAD

**Issue by Aidyn-A** [pytorch/pytorch#89104]
### 🐛 Describe the bug

The following code snippet fails on AMD EPYC 7742 CPU.

```python
import torch

x = torch.rand(1, 32, 512, 512, 256)
m = torch.nn.Conv3d(32, 1, kernel_size=1, padding=0, stride=1, bias=False)
y = m(x)
```

with the error message:
```
  File "test.py", line 5, in <module>
    y = m(x)
  File "/opt/pytorch/pytorch/torch/nn/modules/module.py", line 1423, in _call_impl
    return forward_call(*input, **kwargs)
  File "/opt/pytorch/pytorch/torch/nn/modules/conv.py", line 613, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/opt/pytorch/pytorch/torch/nn/modules/conv.py", line 608, in _conv_forward
    return F.conv3d(
RuntimeError: could not create a primitive
```

OneDNN log:
```
onednn_verbose,info,cpu,runtime:OpenMP,nthr:128
onednn_verbose,info,cpu,isa:Intel AVX2
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:abcde:f0 dst_f32::blocked:aBcde8b:f0,attr-scratchpad:user ,,1x32x512x512x256,9611.53
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:abcde:f0 dst_f32:p:blocked:ABcde8b8a:f0,attr-scratchpad:user ,,1x32x1x1x1,0.00195312
ERROR
```

### Versions

oneDNN v2.7 (650085b2f3643aad05c629425983491d63b5c289)
PyTorch https://github.com/pytorch/pytorch/commit/7ad87f63e248b629d435a199cb61f4ed1f3dfcab and newer

CPU Info:
```
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   43 bits physical, 48 bits virtual
CPU(s):                          256
On-line CPU(s) list:             0-255
Thread(s) per core:              2
Core(s) per socket:              64
Socket(s):                       2
NUMA node(s):                    8
Vendor ID:                       AuthenticAMD
CPU family:                      23
Model:                           49
Model name:                      AMD EPYC 7742 64-Core Processor
Stepping:                        0
Frequency boost:                 enabled
CPU MHz:                         2250.000
CPU max MHz:                     2250.0000
CPU min MHz:                     1500.0000
BogoMIPS:                        4491.83
Virtualization:                  AMD-V
L1d cache:                       4 MiB
L1i cache:                       4 MiB
L2 cache:                        64 MiB
L3 cache:                        512 MiB
NUMA node0 CPU(s):               0-15,128-143
NUMA node1 CPU(s):               16-31,144-159
NUMA node2 CPU(s):               32-47,160-175
NUMA node3 CPU(s):               48-63,176-191
NUMA node4 CPU(s):               64-79,192-207
NUMA node5 CPU(s):               80-95,208-223
NUMA node6 CPU(s):               96-111,224-239
NUMA node7 CPU(s):               112-127,240-255
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP conditional, RSB filling
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_goo
                                 d nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_lega
                                 cy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_ps
                                 tate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_l
                                 lc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pft
                                 hreshold avic v_vmsave_vmload vgif v_spec_ctrl umip rdpid overflow_recov succor smca sme sev sev_es
```

cc @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @VitalyFedyunin @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10 @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen @frank-wei

**Comment by jingxu10** [comment]
this issue is not reproducible on Intel Xeon platforms with 1.13.0 public released wheel.

**Comment by jingxu10** [comment]
We will double confirm on a AMD machine.

**Comment by malfet** [comment]
Added `linunx.2xlarge.amd` runner to be able to run some sort of CI on AMD machines

**Comment by vpirogov** [comment]
Confirmed oneDNN issue with Intel AVX2 codepath (https://github.com/oneapi-src/oneDNN/issues/1500). Processors with Intel AVX-512 support are not impacted.

**Comment by malfet** [comment]
@vpirogov is it possible to force oneDNN to use AVX2 codepath even if CPU supports AVX512?
For example in PyTorch one can test older compute capability codepath by defining `ATEN_CPU_CAPABILITY` variable, see
https://github.com/pytorch/pytorch/blob/74610a1cedbab64e813f3b49535cd8691a3ec5c7/aten/src/ATen/native/DispatchStub.cpp#L13-L15

**Comment by vpirogov** [comment]
@malfet, [`DNNL_MAX_CPU_ISA=AVX2`](https://oneapi-src.github.io/oneDNN/dev_guide_cpu_dispatcher_control.html) will do the trick.

**Comment by vpirogov** [comment]
The fix is released in [oneDNN v2.7.2](https://github.com/oneapi-src/oneDNN/releases/tag/v2.7.2).
